glommio/src/sys/mod.rs

```diff
@@ -11,6 +11,7 @@ use std::{
     ffi::CString,
     fmt,
     io,
+    io::Error,
```

Comment:
again, commenting on the description of the source.
I don't know how big of a change this would be, but the part where you mention references, that would be good for any `Source`.

For consistency, we could add it everywhere. You have this fresh on your mind, how hard would that be ?

The real value of the shared source is that it is not disposed / cancelled when one of them goes away

Reply:
Sorry I got confused (I wrote the commit message after the fact). I'll fix it up. Essentially what I meant to say is that we no longer return a `DmaBuffer` or a `Vec` to consumers. Instead we return the `IoBuffer` directly and it now implements `Deref` so 99% of the time consumers don't even need to `match`, they have easy access to `&[u8]`.

Reply:
The main difference between the two is that a `Source` has one consumer (and therefore, one waker) while `SharedSource` may have many. 

I think it would be pretty simple to unify them at the cost of maybe making `Source` a tad more expensive (even if there's only one waker I'd have to store it in a vec and that's an indirection vs what we currently have).

Reply:
We don't want to make the `Source` more expensive. But couldn't we use traits to represent common behavior ?